core/store/src/trie/mem/arena/alloc.rs

```diff
@@ -77,38 +87,32 @@ impl Allocator {
         self.active_allocs_count_gauge.set(self.active_allocs_count as i64);
         let size_class = allocation_class(size);
         let allocation_size = allocation_size(size_class);
-        if self.freelists[size_class] == usize::MAX {
-            if arena.mmap.len() < self.next_ptr + allocation_size {
-                panic!(
-                    "In-memory trie Arena out of memory; configured as {} bytes maximum,
-                    tried to allocate {} when {} bytes already used",
-                    arena.mmap.len(),
-                    allocation_size,
-                    self.next_ptr
-                );
+        if self.freelists[size_class].is_invalid() {
+            if self.next_ptr.is_invalid()
+                || arena.chunks[self.next_ptr.chunk as usize].len()
```

Comment:
Nit: adding a `fn chunk() -> usize` to `ArenaPos` would slightly reduce the boilerplate and make the code more readable, and similarly for `pos`